src/linux/component.rs

```diff
@@ -1,160 +1,388 @@
+                    .as_path()
+                    .file_name().map(|f| f.is_empty())
+                    .is_some()
+            });
+
+        components.extend(compo);
+        Some(())
+    }
+
+    /// Compute a label out of disponnible informations.
+    /// sysinfo expose a generic interface with a `label`.
+    /// Problem: lot of sensor don't have a label or a device model! ¯\_(ツ)_/¯
+    /// So let's pretend we have a unique label!
+    /// See table in `Component::label` docstring for the table detail.
+    fn format_label(&self, class: &str, id: u32) -> String {
+        let Component {
+            device_model,
+            name,
+            label,
+            ..
+        } = self;
+        let has_label = !label.is_empty();
+        match (has_label, device_model) {
+            (true, Some(device_model)) => {
+                format!("{name} {label} {device_model} {class}{id}")
             }
+            (true, None) => format!("{name} {label}"),
+            (false, Some(device_model)) => format!("{name} {device_model}"),
+            (false, None) => format!("{name} {class}{id}"),
         }
     }
 }
 
-impl Component {
-    /// Creates a new component with the given information.
-    pub(crate) fn new(
-        label: String,
-        input_file: PathBuf,
-        max: Option<f32>,
-        critical: Option<f32>,
-    ) -> Component {
-        let mut c = Component {
-            temperature: 0f32,
-            label,
-            input_file,
-            max: max.unwrap_or(0.0),
-            critical,
-        };
-        c.refresh();
-        c
+impl Default for Component {
+    fn default() -> Self {
+        Component {
+            name: String::default(),
+            label: String::default(),
+            device_model: Option::default(),
+            input_file: PathBuf::new(),
+            highest_file: Option::default(),
+            max: Option::default(),
+            temperature: Option::default(),
+            threshold_max: Option::default(),
+            threshold_min: Option::default(),
+            threshold_critical: Option::default(),
+            sensor_type: Option::default(),
+        }
     }
 }
 
 impl ComponentExt for Component {
+    /// Returns last temperature reading in case of an errors,
+    /// returns `f32::NAN`.
     fn temperature(&self) -> f32 {
-        self.temperature
+        self.temperature.unwrap_or(f32::NAN)
     }
 
+    /// Read current Maximum value encountered may be computed by sysinfo
+    /// or from kernel in case of an errors, returns `f32::NAN`.
     fn max(&self) -> f32 {
-        self.max
+        self.max.unwrap_or(f32::NAN)
     }
 
+    /// Critical threshold defined by chip or kernel may be defined or not.
+    /// Implementation detail:
+    ///
+    /// `thresold_critical` is >= `thresold_max`.
     fn critical(&self) -> Option<f32> {
-        self.critical
+        self.threshold_critical
     }
 
+    /// Label of the termal component.
```

Comment:
Please add or update the documentation on the trait methods declaration directly (under a "linux" section if it's specific to linux).